### PR TITLE
V5 ga 84 dynamic rules editor

### PR DESCRIPTION
**Pull request by SrdjanReblaze**
all fields replaced, except automatic tag fields (missing design for that component)

image

### CODE REVIEW

File: src/views/premium-security/DynamicRulesEditor.vue [hunk 1]
```diff
@@ -815,6 +771,43 @@ export default defineComponent({
...
+
+    prepareTargetOptions() {
+      const options = toRaw(this.targetOptions)
+      const preparedOptions: any[] = []
+
+      options.map((option:any) => preparedOptions.push({
+        option: [option.key, option.title],
+      }))
+
+      console.log(preparedOptions)
```
Comment (Aviv-Galmidi) [inline src/views/premium-security/DynamicRulesEditor.vue:794]: Remove console.log

File: src/views/premium-security/DynamicRulesEditor.vue [hunk 2]
```diff
@@ -815,6 +771,43 @@ export default defineComponent({
...
+
+    prepareCustomResponseNamesOptions() {
+      const options = toRaw(this.customResponseNames)
+      const preparedOptions: any[] = []
+
+      options.map((option:any) => preparedOptions.push({
+        option: [option[0], option[1]],
+      }))
+
+      console.log(preparedOptions)
```
Comment (Aviv-Galmidi) [inline src/views/premium-security/DynamicRulesEditor.vue:807]: Remove console.log

File: src/views/premium-security/DynamicRulesEditor.vue [hunk 3]
```diff
@@ -815,6 +771,43 @@ export default defineComponent({
         },
       )
     },
+
+    prepareNamesOptions() {
```
Comment (Aviv-Galmidi) [inline src/views/premium-security/DynamicRulesEditor.vue:775]: Can these 2 `prepare` functions be turned into computed values?

**Review by Aviv-Galmidi — CHANGES_REQUESTED** [review]

File: src/views/premium-security/DynamicRulesEditor.vue (hunk 3 above)
Comment (SrdjanReblaze) [reply src/views/premium-security/DynamicRulesEditor.vue:770]: done, made different inline solution

File: src/views/premium-security/DynamicRulesEditor.vue (hunk 1 above)
Comment (SrdjanReblaze) [reply src/views/premium-security/DynamicRulesEditor.vue]: done

File: src/views/premium-security/DynamicRulesEditor.vue (hunk 2 above)
Comment (SrdjanReblaze) [reply src/views/premium-security/DynamicRulesEditor.vue]: done

**Review by Aviv-Galmidi — APPROVED** [review]

**Pull request merged**